packages/headless/src/features/analytics/analytics-utils.ts

```diff
@@ -119,6 +126,40 @@ export const noopSearchAnalyticsAction = makeNoopAnalyticsAction(
   AnalyticsType.Search
 );
 
+export const makeCaseAssistAnalyticsAction = (
+  prefix: string,
+  log: (
+    client: CaseAssistClient,
+    state: ConfigurationSection & Partial<CaseAssistAppState>
+  ) => Promise<void | SearchEventResponse> | void
+) => {
+  return createAsyncThunk<
+    {},
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByCaseAssistAnalytics>
+  >(
+    prefix,
+    async (
+      _,
+      {getState, extra: {analyticsClientMiddleware, preprocessRequest, logger}}
+    ) => {
+      const state = getState();
+      const client = configureCaseAssistAnalytics({
+        state,
+        logger,
+        analyticsClientMiddleware,
+        preprocessRequest,
+      });
+      const response = await log(client, state);
```

Comment:
From what I understand if analytics is disabled then we get a disabled client and still execute the log function. What happens then? Is it all no-op in the end?

Reply:
Yes, a no-op client is used when analytics is disabled (see [this file](https://github.com/coveo/coveo.analytics.js/blob/master/src/caseAssist/caseAssistClient.ts#L30)).

Reply:
thanks!